assign.go

```diff
@@ -199,6 +201,28 @@ func assignToSlice(s reflect.Value, jv js.Value) (reflect.Value, error) {
 	return s, nil
 }
 
+func assignToArray(s reflect.Value, jv js.Value) (reflect.Value, error) {
+	t := s.Type()
+	n := jv.Length()
+	et := t.Elem()
+	if !s.CanAddr() {
+		s = reflect.New(reflect.ArrayOf(n, t.Elem())).Elem()
```

Comment:
Removing this `s.CanAddr` check results in `unexpected error: unexpected panic: reflect: reflect.Value.Set using unaddressable value`. I'm not sure if this is the best way to work around the issue though. Open to suggestions.

Reply:
Something feels off about this because arrays can not be nil but it allocations a new array. I'm curious why we need to see if it's addressable, if at all.